### ISSUE TITLE
`%1 is not a valid Win32 application.` when using `wordninja_windows_64_lua53.dll`

### ISSUE THREAD

**Issue by sci-42ver** [BlindingDark/rime-easy-en#48]
I used linux before where I tried `wordninja-rs-lua` or compiling `wordninja-rs` and all works fine.

But when I used weasel recently in `Windows 11 22H2 22621.3155`, I used `rime.lua` to use `easy_en_enhance_filter` and checked in `git-bash`
```bash
$ pwd
/d/Rime/weasel-0.15.0
$ ls lua53.dll data/wordninja_windows_64_lua53.dll
data/wordninja_windows_64_lua53.dll*  lua53.dll*
$ cat /tmp/rime.weasel.DESKTOP-O0JBM8G.czg.log.ERROR.20240314-162538.1228
...
E20240314 16:25:38.838465 19952 lua_gears.cc:59] Lua Compoment of initialize  error:( module: easy_en_enhance_filter name_space: easy_en_enhance_filter status: 2 ): error loading module 'wordninja' from file 'D:\Rime\weasel-0.15.0\data\wordninja_windows_64_lua53.dll':
        %1 is not a valid Win32 application.
E20240314 16:25:47.078372 19952 lua_gears.cc:15] LuaTranslation::Next error(2): D:\Rime\weasel-0.15.0\data\lua\easy_en.lua:59: attempt to call a nil value (upvalue 'wordninja_split')
...
```

I use the 64-bit system, and I tried to redownload the `wordninja_windows_64_lua53.dll` as [this website hint](https://www.computerhope.com/issues/ch000726.htm) but it didn't help.

**Comment by BlindingDark** [comment]
Sorry I don't have a Windows device for testing, but you can try `wordninja_windows_32_lua53.dll` first.

**Comment by sci-42ver** [comment]
> Sorry I don't have a Windows device for testing, but you can try `wordninja_windows_32_lua53.dll` first.

Yes. I also tried this before posting this issue. `wordninja_windows_32_lua53.dll` will cause `weasel` directly crashing without even creating one error log.

Since you don't have one device, _feel free to let this problem be there_. Normally I only needs the feature to add the space after each word since I only use English in most of time. So deleting some codes in the corresponding lua code is enough. I also find one [better solution (although a bit different)](https://github.com/iDvel/rime-ice/issues/327#issuecomment-1579699979) tracking what key is pressed (so we can decide when to avoid the space by pressing one specific key like at the end of one sentence) when I tried the default rime-ice method of iOS `Hamster`.

**Comment by KagaJiankui** [comment]
**Pinpointing the reason by asking Copilot and experimenting**:
  
- The new Rime release integrates **Lua 5.4** as its interpreter.  
- On Windows, both stable and nightly Rust toolchains typically target **x86-64-pc-msvc**.  
- Building therefore requires *explicitly specifying* the **`lua54` feature**.  
- In addition, the appropriate **Lua5.4 MSVC17 dynamic library**(`lua-5.4.2_Win64_dll17_lib.zip`) must be placed into Weasel *program directory* root.

**Summary:** Please ensure the `lua54` feature is specified and the MSVC v17 library is available when building on Windows.  

Try the following build command: 

```bash
export RUSTFLAGS="-C target-cpu=native -C opt-level=3"
git clone https://github.com/BlindingDark/wordninja-rs-lua.git && cd wordninja-rs-lua 
cargo build --release --features "lua54"
```
